https://issues.redhat.com/browse/ACM-12824 (#6730)
open-cluster-management/rhacm-docs: governance/policy_dependencies.adoc

[#policy-dependencies]
= Policy dependencies

Dependencies can be used to activate a policy only when other policies on your cluster are in a certain state. When the dependency criteria is not met, the policy is labeled as `Pending` and resources are not created on your managed cluster. There are more details about the the criteria status in the policy status.

> You can use policy dependencies to control the ordering of how objects are applied. For example, if you have a policy for an operator, and another policy for a resource that operator manages, setting a dependency on the second policy so that it does not attempt to create the resource until the operator is installed could help performance on the managed cluster.

Author: ```suggestion
You can use policy dependencies to control the ordering of how objects are applied. For example, if you have a policy for an operator and another policy for a resource that the operator manages, you can set a dependency on the second policy so that it does not attempt to create the resource until the operator is installed. This can help with the performance on the managed cluster.
```

        remediationAction: enforce
        severity: low
----

> //we need to add the parameters to the sample: extraDependencies and ignorePending | MJ | 07/30

Reviewer: GH won't let me put a suggestion in the right place, but we can start by replacing
```yaml
...
  policy-templates:
  - objectDefinition:
      apiVersion: policy.open-cluster-management.io/v1
...
```
with
```yaml
...
  policy-templates:
  - extraDependencies:
    - apiVersion: policy.open-cluster-management.io/v1
      kind: ConfigurationPolicy
      name: scan-setting-prerequisite
      compliance: Compliant
    ignorePending: false
    objectDefinition:
      apiVersion: policy.open-cluster-management.io/v1
...
```

*Note:* A dependency cannot be used to apply a policy on one cluster based on the status of a policy in another cluster. 
<1> The `dependencies` field is set on a `Policy` object, and the requirements apply to all policy templates in the policy. 

> <2> The `extraDependencies` field can be set on individual policy template. For example the parameter can be set in a configuration policy, and defines criteria that must be satisfied in addition to any `dependencies` set in the policy.

Reviewer: ```suggestion
<2> The `extraDependencies` field can be set on individual policy template. For example the parameter can be set for a configuration policy, and defines criteria that must be satisfied in addition to any `dependencies` set in the policy.
```

> <3> The `ignorePending` field can be set on individual policy templates, and configures whether the `Pending` status on a template is considered as `Compliant` or `NonCompliant` when the overall policy compliance is calculated. By default, this is set to `false`, and a `Pending` template means that the policy is `NonCompliant`. When you set this to `true` it is useful when the expected status of the template `Pending`.

Reviewer: I'm still thinking about the best way to describe this... sorry it's a complicated one :grimacing:  
```suggestion
<3> The `ignorePending` field can be set each individual policy template, and configures whether the `Pending` status on that template is considered as `Compliant` or `NonCompliant` when the overall policy compliance is calculated. By default, this is set to `false`, and a `Pending` template causes the policy status to be `NonCompliant`. When you set this to `true`, the policy can still be `Compliant` when this template is `Pending`, which is particularly useful when that is expected status of the template.
```

        remediationAction: enforce
        severity: low
----
<1> The `dependencies` field is set on a `Policy` object, and the requirements apply to all policy templates in the policy. 
<2> The `extraDependencies` field can be set on individual policy template. For example the parameter can be set for a configuration policy, and defines criteria that must be satisfied in addition to any `dependencies` set in the policy.
<3> The `ignorePending` field can be set on each individual policy template, and configures whether the `Pending` status on that template is considered as `Compliant` or `NonCompliant` when the overall policy compliance is calculated. By default, this is set to `false` and a `Pending` template causes the policy to be `NonCompliant`. When you set this to `true` the policy can still be `Compliant` when this template is `Pending`, which is useful when that is expected status of the template.

> *Note:* A dependency cannot be used to apply a policy on one cluster based on the status of a policy in another cluster.

Reviewer 2: I think we can change this to active voice

> *Note:* A dependency cannot be used to apply a policy on one cluster based on the status of a policy in another cluster.

Author: ```suggestion
*Note:* You cannot use a dependency to apply a policy on one cluster based on the status of a policy in another cluster.
```
